core/unit_test/incremental/Test03a_ParallelFor.hpp

```diff
@@ -0,0 +1,152 @@
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <Kokkos_Core.hpp>
+#include <cstdio>
+#include <sstream>
+#include <type_traits>
+#include <gtest/gtest.h>
+
+/// @Kokkos_Feature_Level_Required:3
+
+namespace Test {
+
+using DataType       = double;
+const DataType value = 0.5;
+
+// parallel-for unit test
+// create an functor which multiplies the index with a constant double value
+
+struct Functor {
+  DataType *_data;
+
+  Functor(DataType *data) : _data(data) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i) const { _data[i] = i * value; }
+};
+
+template <class ExecSpace>
+struct TestParallel_For {
+  int num_elements = 10;
+  DataType *DeviceData, *HostData;
+
+  // memory_space for the memory allocation
+  typedef typename TEST_EXECSPACE::memory_space MemSpaceD;
+  typedef Kokkos::HostSpace MemSpaceH;
+
+  void compare_equal(DataType *data) {
+    for (int i = 0; i < num_elements; ++i) {
+      ASSERT_EQ(data[i], i * value);
+    }
+  }
+
+  // A simple parallel for test with functors
+  void simple_test() {
+    // Allocate Memory for both device and host memory spaces
+    DeviceData = (DataType *)Kokkos::kokkos_malloc<MemSpaceD>(
+        "dataD", num_elements * sizeof(DataType));
+    HostData = (DataType *)Kokkos::kokkos_malloc<MemSpaceH>(
+        "dataH", num_elements * sizeof(DataType));
+
+    // parallel-for functor called for num_elements elements
+    Kokkos::parallel_for("parallel_for", num_elements, Functor(DeviceData));
+
+    // Copy the data back to Host memory space
+    Kokkos::Impl::DeepCopy<MemSpaceD, MemSpaceH>(
+        HostData, DeviceData, num_elements * sizeof(DataType));
+
+    // Check if all data has been update correctly
+    compare_equal(HostData);
+
+    // Free the allocated memory
+    Kokkos::kokkos_free<MemSpaceD>(DeviceData);
+    Kokkos::kokkos_free<MemSpaceH>(HostData);
+  }
+
+  // A parallel_for test with user defined RangePolicy
+  void range_policy() {
+    // Create a range policy for the parallel_for
+#if defined(KOKKOS_ENABLE_CUDA)
```

Comment:
Why do you need the distinction?

Reply:
I just kept that to test the schedule type feature. Not doing anything specific with it.